Add --overwrite option to rnp

ni4 opened the pull request:
Before now rnp was overwriting output file by default, and without any option to change this behavior.
So this PR:
- adds `--overwrite` option, which will allow to overwrite output by default
- fixes weird double fd opening during file signing (should be typo)
- updates handling of output stream creation with asking user whether to overwrite file/new filename

flowher commented:
"Before now rnp was overwriting output file by default", is it really true? Or do you mean that it wasn't overwriting?

flowher commented on src/rnp/rnpcfg.c:
```diff
@@ -306,17 +306,12 @@ bool
-    } else {
-        return false;
+    
+    if (val && ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0) || (atoi(val) > 0))) {
```
better to use ``tolower()`` and then compare to "true"

ronaldtse replied:
Agree.

ni4 replied:
Used strcasecmp - otherwise would need to check val, allocate buffer for lowercase copy, deallocate, etc, etc.

flowher commented on src/rnp/rnpcfg.c:
```diff
@@ -306,17 +306,12 @@ bool
-    } else {
-        return false;
+    
+    if (val && ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0) || (atoi(val) > 0))) {
```
easier to read if "return (val && ((strcmp(val, "true") == 0) || (strcmp(val, "True") == 0) || (atoi(val) > 0))) "

ronaldtse replied:
Agree.

ni4 replied:
Done.

flowher commented on src/lib/misc.c:
```diff
@@ -1134,3 +1134,15 @@ rnp_strlwr(char *s)
+    }
+
+    return s;
+}
```
\n

ni4 replied:
Ok

flowher commented on src/lib/rnp.c:
```diff
@@ -1200,11 +1200,116 @@ rnp_decrypt_file(rnp_ctx_t *ctx, const char *f, const char *out)
+rnp_initialize_output(rnp_ctx_t *ctx, pgp_dest_t *dst, const char *out)
+{
+    char newname[PATH_MAX];
+    bool is_stdout = out == NULL || out[0] == '\0' || strcmp(out, "-") == 0;
```
can you put () around those conditions. It hard to read this way

ni4 replied:
Ok.

flowher commented:
LGTM, some minor comments added

ni4 commented:
@flowher @ronaldtse Thanks for comments, fixed.

> "Before now rnp was overwriting output file by default", is it really true? Or do you mean that it wasn't overwriting?

Yeah, it was. overwriting without the prompt. It has hardcoded setting overwrite to 1, so I moved the same to rnpcfg that time.

ni4 commented:
@flowher Any other comments, or this may be merged?

flowher commented on src/lib/misc.c:
```diff
@@ -1134,3 +1134,15 @@ rnp_strlwr(char *s)
+{
+    size_t len = strlen(s);
+
+    while (len > 0 && (s[len - 1] == '\n' || s[len - 1] == '\r')) {
```
() here also please

ni4 replied:
Done

flowher commented on include/rnp/rnp_sdk.h:
```diff
@@ -71,4 +71,6 @@ bool rnp_hex_decode(const char *hex, uint8_t *buf, size_t buf_len);
 
 char *rnp_strlwr(char *s);
 
+char *rnp_strip_eol(char *s);
```
Please add comment what this function really does

ni4 replied:
Done

flowher commented:
Two additional comments. Go ahead and merge after they're done
